src/internal/task/util.go

```diff
@@ -45,3 +48,62 @@ func DoBatch(ctx context.Context, doer Doer, inputs []*types.Any, cb CollectFunc
 	})
 	return errors.EnsureStack(eg.Wait())
 }
+
+func translateTaskState(state State) taskapi.State {
+	switch state {
+	case State_RUNNING:
+		return taskapi.State_RUNNING
+	case State_SUCCESS:
+		return taskapi.State_SUCCESS
+	case State_FAILURE:
+		return taskapi.State_FAILURE
+	}
+	return taskapi.State_UNKNOWN
+}
+
+// HandleList implements the functionality for an arbitrary service's ListTask gRPC
+func HandleList(ctx context.Context, svc Service, req *taskapi.ListTaskRequest, send func(info *taskapi.TaskInfo) error) error {
+	var marshaler jsonpb.Marshaler
+	return errors.EnsureStack(svc.List(ctx, req.Group.Namespace, req.Group.Group, func(key *TaskKey, data *Task, claimed bool) error {
+		state := translateTaskState(data.State)
+		if claimed {
+			state = taskapi.State_CLAIMED
+		}
+		var input types.DynamicAny
+		if err := types.UnmarshalAny(data.Input, &input); err != nil {
+			return errors.EnsureStack(err)
+		}
+		inputJSON, err := marshaler.MarshalToString(input.Message)
+		if err != nil {
+			return errors.EnsureStack(err)
+		}
+		info := &taskapi.TaskInfo{
+			ID: data.ID,
+			Group: &taskapi.Group{
```

Comment:
Very minor nit, but I think we should in general initialize in order with respect to the order that the fields are defined. So, namespace, then group.